src/document/Document.js

```diff
@@ -675,10 +675,12 @@ define(function (require, exports, module) {
     
     /**
      * Updates the language to match the current mapping given by LanguageManager
+     * 
+     * @param {!string} rawText  Text content of the file.
      */
-    Document.prototype._updateLanguage = function () {
+    Document.prototype._updateLanguage = function (rawText) {
```

Comment:
But also... since this is called in multiple cases, various actions will trigger a "re-check" of the document's text.  If the user edits the document to add/remove the shebang, at what point would they expect the document to switch to/from Bash mode?  We should make sure that feels predictable.